celery/app/amqp.py

```diff
@@ -589,6 +590,15 @@ def router(self):
     def router(self, value):
         return value
 
+    @cached_property
+    def repr(self):
+        repr_function = self.app.conf.task_args_repr_function
+        if not repr_function:
+            return lambda o, **kwargs: o
+        elif isinstance(repr_function, str):
+            return mlazy(symbol_by_name, repr_function)
```

Comment:
We should test if `repr_function` is a callable.
If it isn't this should also raise an `ImproperlyConfigured` exception.